### ISSUE TITLE
Duplicate weight initialization

### ISSUE THREAD

**Issue by JIElite** [wing3s/pysc2-rl-mini#1]
[Question1]
In model.py, [line 107 - 117](https://github.com/wing3s/pysc2-rl-mini/blob/master/model.py#L107):
The following layer's weight was initalized by xavier init but at line 111 they are also initialized by weight_norm. Why do we initialize their weights twice?
- ns_fc3
- nsa_fc4
- nsc_fc4

[Question2]
In the same file, line 119:
Why do we need to add this line: ```self.train()```. According to [pytorch official document](http://pytorch.org/docs/master/nn.html#torch.nn.Module.train), it says:

> This has any effect only on modules such as Dropout or BatchNorm.

Does this mean we can remove this line?

Thank you, this repo is awesome and with nice coding structure.

**Comment by wing3s** [comment]
For Q1, I was following weight initialization approaches from [ikostrikov/pytorch-a3c/model.py](https://github.com/ikostrikov/pytorch-a3c/blob/895c82a832f349be9f4222a07895d97386b4cea5/model.py#L45). It actually does twice weight initializations. I haven't done comparing performance between different initialization approaches. I've refactored to cleaner solution https://github.com/wing3s/pysc2-rl-mini/commit/c5551fbe26e82c50cc8ef16a7cd87f3875fa91fa without stacking initializations.

For Q2, from source code, `self.train()` only has impact on Dropout and BatchNorm like you mentioned. Theoretically, we can remove it. I haven't done testing removing these lines to prove it.

Thanks for digging into the code. I'll circle back to these questions once I'm in final testing phase.

**Comment by JIElite** [comment]
Thank you for your response :)
